File: app/src/main/java/de/kuschku/quasseldroid_ng/util/accounts/Account.java
/*
 * QuasselDroid - Quassel client for Android
 * Copyright (C) 2016 Janne Koschinski
 * Copyright (C) 2016 Ken Børge Viktil
 * Copyright (C) 2016 Magnus Fjell
 * Copyright (C) 2016 Martin Sandsmark <martin.sandsmark@kde.org>
 *
 * This program is free software: you can redistribute it and/or modify it
 * under the terms of the GNU General Public License as published by the Free
 * Software Foundation, either version 3 of the License, or (at your option)
 * any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package de.kuschku.quasseldroid_ng.util.accounts;

import java.util.UUID;

import de.kuschku.util.ServerAddress;

public class Account {
    public final UUID id;

    public final String name;

    public final String host;
    public final int port;

    public final String user;
    public final String pass;

    public Account(UUID id, String name, String host, int port, String user, String pass) {
        this.id = id;
        this.name = name;
        this.host = host;
        this.port = port;
        this.user = user;
        this.pass = pass;
    }

    public ServerAddress toAddress() {
        return new ServerAddress(host, port);
    }
}

File: persistence/src/main/java/de/kuschku/quasseldroid/persistence/models/Account.kt
/*
 * Quasseldroid - Quassel client for Android
 *
 * Copyright (c) 2020 Janne Mareike Koschinski
 * Copyright (c) 2020 The Quassel Project
 *
 * This program is free software: you can redistribute it and/or modify it
 * under the terms of the GNU General Public License version 3 as published
 * by the Free Software Foundation.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */

package de.kuschku.quasseldroid.persistence.models

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import de.kuschku.quasseldroid.persistence.util.AccountId

@Entity(tableName = "Account")
data class Account(
  @PrimaryKey(autoGenerate = true)
  @ColumnInfo(name = "id")
  var rawId: Long,
  var host: String,
  var port: Int,
  var requireSsl: Boolean,
  var user: String,
  var pass: String,
  var name: String,
  var lastUsed: Long,
  var acceptedMissingFeatures: Boolean,
  var defaultFiltered: Int
) {
  inline var id
    get() = AccountId(rawId)
    set(value) {
      rawId = value.id
    }

  companion object {
    inline fun of(
      id: AccountId,
      host: String,
      port: Int,
      requireSsl: Boolean,
      user: String,
      pass: String,
      name: String,
      lastUsed: Long,
      acceptedMissingFeatures: Boolean,
      defaultFiltered: Int
    ) = Account(
      id.id,
      host,
      port,
      requireSsl,
      user,
      pass,
      name,
      lastUsed,
      acceptedMissingFeatures,
      defaultFiltered
    )
  }
}
